go/arrow/csv/reader_test.go

```diff
@@ -350,10 +358,12 @@ rec[2]["f32"]: [(null)]
 rec[2]["f64"]: [(null)]
 rec[2]["str"]: [(null)]
 rec[2]["ts"]: [(null)]
+rec[2]["list(i64)"]: [(null)]
+rec[2]["uuid"]: (extension_type<storage=fixed_size_binary[16]>)[(null)]
 `
-
-	if got, want := out.String(), want; got != want {
-		t.Fatalf("invalid output:\ngot= %s\nwant=%s\n", got, want)
+	got, want := out.String(), want
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Fatalf("invalid output: (-want, +got) = %s", diff)
```

Comment:
instead of adding a new dependency, we could use `github.com/stretchr/testify/assert` which will also create the diff for you as you need by using `assert.Equal`